ores/score_processors/score_processor.py

```diff
@@ -20,7 +20,7 @@ def __init__(self, scoring_contexts, score_cache=None,
         self.metrics_collector = metrics_collector or Null()
 
     def score(self, context, model, rev_ids, caches=None,
-              include_features=False, precache=False):
+              include_features=False, precache=False, features_cache={}):
```

Comment:
I think you want to make use of "caches" since those turn into the feature caches.  It's a mapping of `rev_id`-->(`dependent`-->`value`).  Each revision will need to have it's own cache.